dot/parachain/prospective-parachains/messages.go

```diff
@@ -0,0 +1,173 @@
+package prospective_parachains
+
+import (
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+type ProspectiveParachainMessageValues interface {
+	CandidateBacked | GetBackableCandidates | GetHypotheticalMembership |
+		GetMinimumRelayParents
+}
+
+// ProspectiveParachainsMessage Messages sent to the Prospective Parachains subsystem.
+type ProspectiveParachainsMessage interface {
+	isProspectiveParachainsMessage()
+}
+
+// IntroduceSecondedCandidate Inform the Prospective Parachains Subsystem of a new seconded candidate.
+// The response is false if the candidate was rejected by prospective parachains,
+// true otherwise (if it was accepted or already present)
+type IntroduceSecondedCandidate struct {
+	IntroduceSecondedCandidateRequest
+	Response chan bool
+}
+
+func (IntroduceSecondedCandidate) isProspectiveParachainsMessage() {}
+
+// IntroduceSecondedCandidateRequest Request introduction of a seconded candidate into the prospective parachains
+// subsystem.
+type IntroduceSecondedCandidateRequest struct {
+	// The para-id of the candidate.
+	CandidateParaID parachaintypes.ParaID
+	// The candidate receipt itself.
+	CandidateReceipt parachaintypes.CommittedCandidateReceipt
+	// The persisted validation data of the candidate.
+	PersistedValidationData parachaintypes.PersistedValidationData
+}
+
+// CandidateBacked Inform the Prospective Parachains Subsystem that a previously introduced candidate
+// has been backed. This requires that the candidate was successfully introduced in
+// the past.
+type CandidateBacked struct {
+	ParaId        parachaintypes.ParaID
+	CandidateHash parachaintypes.CandidateHash
+}
+
+func (CandidateBacked) isProspectiveParachainsMessage() {}
+
+// GetBackableCandidates Try getting requested quantity of backable candidate hashes along with their relay parents
+// for the given parachain, under the given relay-parent hash, which is a descendant of the given ancestors.
+// Timed out ancestors should not be included in the collection.
+// RequestedQty should represent the number of scheduled cores of this ParaId.
+// A timed out ancestor frees the cores of all of its descendants, so if there's a hole in the
+// supplied ancestor path, we'll get candidates that backfill those timed out slots first. It
+// may also return less/no candidates, if there aren't enough backable candidates recorded.
+type GetBackableCandidates struct {
+	RelayParentHash common.Hash
+	ParaId          parachaintypes.ParaID
+	RequestedQty    uint32
+	Ancestors       Ancestors
```

Comment:
Do we really need to have type declare for Ancestors? Can we use the slice of hash type directly here?
```suggestion
	// Ancestors A collection of ancestor candidates of a parachain.
	Ancestors       []parachaintypes.CandidateHash
```

Reply:
Updated ancestors to a map.